T/TorchCAPI/build_tarballs.jl

```diff
@@ -0,0 +1,114 @@
+using BinaryBuilder, Pkg
+using BinaryBuilderBase: get_addable_spec
+
+include(joinpath(@__DIR__, "..", "..", "fancy_toys.jl"))
+include(joinpath(@__DIR__, "..", "..", "platforms", "cuda.jl"))
+
+name = "TorchCAPI"
+version = v"0.2.0"
+
+torch_version = v"1.10.2"
+
+sources = [
+    GitSource("https://github.com/FluxML/Torch.jl.git", "d1711d716c4993ca25e975aad5f7a638cfa7d7c2"),
+]
+
+script = raw"""
+cmake_extra_args=""
+if [[ $bb_full_target == *cuda* ]]; then
+    # CMake toolchain looks for compiler in CUDA_PATH/bin/nvcc
+    export CUDA_PATH="$prefix/cuda"
+    if [[ $bb_full_target == *cuda+11.3* ]]; then
+        export CUDARTLIB=cudart
+        export cmake_extra_args="\
+            -DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY \
+            -DCMAKE_CUDA_COMPILER_ID_RUN=1"
+    fi
+    cmake_extra_args+="\
+        -DUSE_CUDA=ON \
+        -DCUDA_TOOLKIT_ROOT_DIR=$CUDA_PATH"
+else
+    cmake_extra_args+="\
+        -DUSE_CUDA=OFF"
+fi
+
+cd Torch.jl
+install_license LICENSE
+
+cd deps/c_wrapper
+configure() {
+    cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=$prefix \
+        -DCMAKE_TOOLCHAIN_FILE=$CMAKE_TARGET_TOOLCHAIN \
+        $cmake_extra_args \
+        -S . \
+        -B build
+}
+configure || configure || configure
+
+if [[ $bb_full_target == aarch64-linux-gnu-*-cxx03 ]]; then # Somehow CMake ends up adding '-D_GLIBCXX_USE_CXX11_ABI=' to CXX_FLAGS
+    sed -i 's/ -D_GLIBCXX_USE_CXX11_ABI= / -D_GLIBCXX_USE_CXX11_ABI=0 /' build/CMakeFiles/torch_c_api.dir/flags.make
+fi
+
+cmake --build build -- -j $nproc
+
+install -Dvm 755 build/libtorch_c_api.$dlext $libdir/libtorch_c_api.$dlext
+
+for header_path in $(ls -1 *.h | grep -v cpp); do
```

Comment:
https://mywiki.wooledge.org/ParsingLs

Reply:
Just a hack to get around a “.cpp.h”-file which should not be copied - it’s implementation.

Reply:
I was suggesting not to use `ls`.

Reply:
Yeah, I know - but not sure what would be better here? Just get rid of the loop and call install twice?

There are just these specific .h-files present:
```
torch_api.h
torch_api_generated.h
```
and then the silly `torch_api_generated.cpp.h`; Which I will fix in the next iteration in [deps/c_wrapper_generator/bin/main.ml#L282](https://github.com/FluxML/Torch.jl/blob/master/deps/c_wrapper_generator/bin/main.ml#L282)

Reply:
Can we go with the lazy/easy solution here (no change - cf. edit 2)?